core/gesture_handler.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2017 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview The class that handlers gestures (drags, clicks, etc) across
+ * all of Blockly, including multiple workspaces.
+ * @author fenichel@google.com (Rachel Fenichel)
+ */
+'use strict';
+
+/**
+ * @name Blockly.GestureHandler
+ * @namespace
+ */
+goog.provide('Blockly.GestureHandler');
+
+goog.require('Blockly.Gesture');
+goog.require('Blockly.Touch');
+
+
+/**
+ * Map from touch identifiers to in-process gestures.
+ * @type {!Object<string, Blockly.Gesture>}
+ * @private
+ */
+Blockly.GestureHandler.gestureMap_ = Object.create(null);
+
+/**
+ * How many gestures are currently being tracked.
+ * @type {number}
+ * @private
+ */
+Blockly.GestureHandler.numGesturesInProgress_ = 0;
+
+/**
+ * The maximum allowable number of concurrent gestures.
+ * @type {number}
+ * @const
+ */
+Blockly.GestureHandler.MAX_GESTURES = 1;
+
+/**
+ * Look up the gesture that is tracking this touch stream.  May create a new
+ * gesture.
+ * @param {!Event} e Mouse event or touch event
+ * @return {Blockly.Gesture} The gesture that is tracking this touch stream,
+ *     or null if no valid gesture exists.
+ */
+Blockly.GestureHandler.gestureForEvent = function(e) {
```

Comment:
Are all of these methods intentionally static (not sure that's the right word in javascript)? In other words, is a GestureHandler not something you expect to have to construct and so anywhere in the code base or any developer can just call gestureForEvent, removeGestureForId, etc?